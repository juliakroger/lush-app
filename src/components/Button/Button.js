import React from 'react'
import PropTypes from 'prop-types'

import styles from './Button.module.scss'

const Button = ({ text, type, clickFunction, id }) => {
  return (
    <button className={styles[type]} onClick={clickFunction} id={id}>
      {text}
    </button>
  )
}

Button.propTypes = {
  text: PropTypes.string,
  type: PropTypes.oneOf([
    'buttonDark',
    'buttonLight',
    'buttonRed',
    'buttonMint'
  ]),
  clickFunction: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired
}

Button.defaultProps = {
  text: '',
  type: 'buttonDark'
}

export default Button
