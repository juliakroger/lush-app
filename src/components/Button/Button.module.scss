@import '../../scss/variables.scss';
@import '../../scss/responsive-utils';

@mixin simpleButton {
  font-size: $font-size-small;
  margin: 10px;
  color: $white-color;
  width: 100%;
  padding: 10px 15px;
  text-align: center;
  border-radius: 4px;
  cursor: pointer;
  text-transform: uppercase;
}

.buttonDark {
  @include simpleButton();
  background-color: $light-black-color;
  border: 1px solid $light-black-color;
}

.buttonLight {
  @include simpleButton();
  background-color: $grey-color;
  border: 1px solid $grey-color;
  color: $black-color;
}

.buttonRed {
  @include simpleButton();
  background-color: $super-orange-color;
  border: 1px solid $super-orange-color;
}

.buttonMint {
  @include simpleButton();
  background-color: $mint-color;
  border: 1px solid $mint-color;
  color: $black-color;
}
