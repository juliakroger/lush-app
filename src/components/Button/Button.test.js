import React from 'react'
import { shallow } from 'enzyme'
import Button from './Button'

describe('Component - Button', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <Button text='my button' clickFunction={() => {}} id='#test-button' />
    )
    expect(component).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    const component = shallow(
      <Button
        text='my button'
        clickFunction={() => {}}
        id='#test-button'
        type='buttonLight'
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    const component = shallow(
      <Button
        text='my button'
        clickFunction={() => {}}
        id='#test-button'
        type='buttonRed'
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('should render without crashing', () => {
    const component = shallow(
      <Button
        text='my button'
        clickFunction={() => {}}
        id='#test-button'
        type='buttonMint'
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('should trigger change without crashing', () => {
    const mockFunc = jest.fn()

    const component = shallow(
      <Button
        text='my button'
        clickFunction={mockFunc}
        id='#test-button'
        type='buttonMint'
      />
    )

    component.find('button').simulate('click')
    expect(mockFunc.mock.calls.length).toBe(1)
  })
})
