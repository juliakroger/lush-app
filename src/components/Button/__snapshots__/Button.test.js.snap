// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Button should render without crashing 1`] = `
<button
  className="buttonDark"
  id="#test-button"
  onClick={[Function]}
>
  my button
</button>
`;

exports[`Component - Button should render without crashing 2`] = `
<button
  className="buttonLight"
  id="#test-button"
  onClick={[Function]}
>
  my button
</button>
`;

exports[`Component - Button should render without crashing 3`] = `
<button
  className="buttonRed"
  id="#test-button"
  onClick={[Function]}
>
  my button
</button>
`;

exports[`Component - Button should render without crashing 4`] = `
<button
  className="buttonMint"
  id="#test-button"
  onClick={[Function]}
>
  my button
</button>
`;
